In [1]:
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt

In [2]:
df = pd.read_csv('../data/WineQT.csv')
df.head()

,fixed acidity,volatile acidity,citric acid,residual sugar,chlorides,free sulfur dioxide,total sulfur dioxide,density,pH,sulphates,alcohol,quality,Id
0,7.4,0.70,0.00,1.9,0.076,11.0,34.0,0.9978,3.51,0.56,9.4,5,0
1,7.8,0.88,0.00,2.6,0.098,25.0,67.0,0.9968,3.20,0.68,9.8,5,1
2,7.8,0.76,0.04,2.3,0.092,15.0,54.0,0.9970,3.26,0.65,9.8,5,2
3,11.2,0.28,0.56,1.9,0.075,17.0,60.0,0.9980,3.16,0.58,9.8,6,3
4,7.4,0.70,0.00,1.9,0.076,11.0,34.0,0.9978,3.51,0.56,9.4,5,4


In [3]:
df.shape

(1143, 13)

In [4]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 1143 entries, 0 to 1142
Data columns (total 13 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   fixed acidity         1143 non-null   float64
 1   volatile acidity      1143 non-null   float64
 2   citric acid           1143 non-null   float64
 3   residual sugar        1143 non-null   float64
 4   chlorides             1143 non-null   float64
 5   free sulfur dioxide   1143 non-null   float64
 6   total sulfur dioxide  1143 non-null   float64
 7   density               1143 non-null   float64
 8   pH                    1143 non-null   float64
 9   sulphates             1143 non-null   float64
 10  alcohol               1143 non-null   float64
 11  quality               1143 non-null   int64  
 12  Id                    1143 non-null   int64  
dtypes: float64(11), int64(2)
memory usage: 116.2 KB


In [5]:
df.isnull().sum()

fixed acidity           0
volatile acidity        0
citric acid             0
residual sugar          0
chlorides               0
free sulfur dioxide     0
total sulfur dioxide    0
density                 0
pH                      0
sulphates               0
alcohol                 0
quality                 0
Id                      0
dtype: int64

In [6]:
df.duplicated().sum()

np.int64(0)

In [7]:
df['Id'].value_counts()

Id
0       1
1       1
2       1
3       1
4       1
       ..
1592    1
1593    1
1594    1
1595    1
1597    1
Name: count, Length: 1143, dtype: int64

In [17]:
df['quality'].value_counts()

quality
5    483
6    462
7    143
4     33
8     16
3      6
Name: count, dtype: int64

In [16]:
df.describe().T

,count,mean,std,min,25%,50%,75%,max
fixed acidity,1143.0,8.311111,1.747595,4.60000,7.10000,7.90000,9.100000,15.90000
volatile acidity,1143.0,0.531339,0.179633,0.12000,0.39250,0.52000,0.640000,1.58000
citric acid,1143.0,0.268364,0.196686,0.00000,0.09000,0.25000,0.420000,1.00000
residual sugar,1143.0,2.532152,1.355917,0.90000,1.90000,2.20000,2.600000,15.50000
chlorides,1143.0,0.086933,0.047267,0.01200,0.07000,0.07900,0.090000,0.61100
free sulfur dioxide,1143.0,15.615486,10.250486,1.00000,7.00000,13.00000,21.000000,68.00000
total sulfur dioxide,1143.0,45.914698,32.782130,6.00000,21.00000,37.00000,61.000000,289.00000
density,1143.0,0.996730,0.001925,0.99007,0.99557,0.99668,0.997845,1.00369
pH,1143.0,3.311015,0.156664,2.74000,3.20500,3.31000,3.400000,4.01000
sulphates,1143.0,0.657708,0.170399,0.33000,0.55000,0.62000,0.730000,2.00000


In [15]:
print('Коэффиценты вариативности:')
for col in df.columns:
    print(f'{col} - {round((df[col].std() / df[col].mean()) * 100, 2)}%')  

Коэффиценты вариативности:
fixed acidity - 21.03%
volatile acidity - 33.81%
citric acid - 73.29%
residual sugar - 53.55%
chlorides - 54.37%
free sulfur dioxide - 65.64%
total sulfur dioxide - 71.4%
density - 0.19%
pH - 4.73%
sulphates - 25.91%
alcohol - 10.36%
quality - 14.24%
Id - 57.64%


## Размер
- 1143 строки, 13 столбцов

## Нули
- Явных нулей нет

## Дубликаты
- Дубликаты не обнаружены

# Столбцы
- `fixed acidity` - фиксированная кислотность. Тип данных - float. Является вещественным числовым непрерывным признаком. Находится в диапазоне от 4.6 до 15.9. Среднее значение 8.31, медианное 7.9. Наблюдается небольшой перекос вправо. Стандартное отклонение 1.74, что говорит об умеренном разбросе значений признака. Коэффицент вариативности в 21% подтверждает умеренную изменчивость признака. Большая часть значений сосредоточена вблизи среднего, однако присутствует определенная вариативность.
- `volatile acidity` - летучая кислотность. Тип данных float. Является вещественным числовым непрерывным признаком. Находится в диапазоне от 0.12 до 1.58. Среднее значение 0.53, медианное 0.52. Распределение близко к симметричному, наблюдается слабый перекос вправо. Стандартное отклонение 0.17, коэффицент вариативности - 33.81%, что свидетельствует о высокой вариации признака.
- `citric acid` - лимонная кислота. Тип данных - float. Является вещественным числовым непрерывным признаком. Находится в диапазоне от 0 до 1. Среднее значение 0.27, медианное 0.25. Наблюдается слабый перекос вправо, хотя распределение близко к симметричному. Стандартное отклонение 0.19 говорит об очень высокой вариативности. Коэффицент вариации 73%.
- `residual sugar` - остаточный сахар. Тип данных - float. Является вещественным числовым непрерывным признаком. Находится в диапазоне от 0.9 до 15.5. Среднее значение 2.5, медианное 2.2, что говорит об умеренном перекосе вправо. Значения признака слабо ассиметричны. Стандартное отклонение 1.35, что говорит об сильной вариации признака. Коэффицент вариации - 53.5%. Большая часть значений сосредоточена вблизи среднего.
- `chlorides` - хлориды. Тип данных - float. Является вещественным числовым непрерывным признаком. Находится в диапазоне от 0.01 до 0.61. Среднее значение 0.0869, медианное 0.079. Наблюдается очень слабый перекос вправо, распределение близко к симметричному. Стандартное отклонение 0.04, что говорит о достаточно высокой вариативности значений признака, хотя большая часть значений расположена вблизи среднего. Коэффицент вариации 54.37%.
- `free sulfur dioxide` - свободный диоксид серы. Тип данных - float. Является вещественным числовым непрерывным признаком. Находится в диапазоне от 1 до 68. Среднее значение 15.61, медианное 13, что говорит об заметном перекосе вправо. Значения признака ассиметричны, есть выраженный хвост вправо. Стандартное отклонение 10.25, что говорит об сильной вариативности данных. Коэффицент вариации 65.64%. 
- `total sulfur dioxide` - общий диоксид серы. Тип данных - float. Является вещественным числовым непрерывном признаком. Находится в диапазоне от 6 до 289. Среднее 45.91, медианное 37, что говорит об сильном перекосе вправо. Скорее всего есть ярко выраженный хвост справа. Стандартное отклоенение 32.78, что говорит об сильной вариативности значений признака. Коэффицент вариации 71.4%.
- `density` - плостность вина. Тип данных float. Является вещественным числовым непрерывным признаком. Находится в диапазоне от 0.99007 до 1.00369. Среднее 0.9967, медианное 0.9978, что говорит об слабом перекосе влево. Стандартное отклонение 0.001 и коэффициент вариации 0.19% свидетельствуют об очень низкой вариативности признака: значения сильно сосредоточены вокруг среднего.
- `pH` - баланс кислот и щелочей. Тип данных float. Является вещественным числовым непрерывным признаком. Находится в диапазоне от 2.74 до 4.01. Если судить по смыслу, в этом столбце оценивается именно сила кислоты. Среднее значение 3.311, медианное 3.31. Есть очень слабый перекос вправо, но в целом распределение очень близко к симметричному. Стандартное отклонение 0.15, коэффицент вариации 4.73%. Значения признака очень слабо вариативны. Значения сильно сконцентрированы вокруг среднего.
- `sulphates` - сульфаты. Тип данных float. Является вещественным числовыми непрерывным признаком. Находится в диапазоне от 0.33 до 2. Среднее 0.65, медиана 0.62. Есть слабый перекос значений вправо. Стандартное отклонение 0.17, коэффицент вариации 26%, что говорит об умеренной вариации данных. Но большая часть данных расположена вблизи среднего.
- `alcohol` - алкоголь. Тип данных float. Является вещественным числовым непрерывным признаком. Находится в диапазоне от 8.4 до 14.9. Среднее значение 10.442, медианное 10.2, что говорит об слабом перекосе вправо. Стандартное отклонение 1.08, коэффицент вариации 10.36%, что говорит об слабой вариации значений. 
- `quality` — оценка качества вина. **Целевой признак**. Тип данных — int. Является порядковым (ordinal) дискретным признаком. Значения находятся в диапазоне от 3 до 8. Распределение классов несбалансированное: преобладают оценки 5 и 6 (более 450 наблюдений), также присутствует умеренное количество вин с оценкой 7. Редко встречаются оценки 3, 4 и 8 (6, 33 и 16 наблюдений соответственно). Данный дисбаланс классов может привести к снижению качества предсказания редких классов при обучении моделей классификации.
- `Id` - идентификатор каждого вина. Повторяющихся значений нет. Признак не несёт информативной нагрузки для модели и не содержит информации, полезной для прогнозирования целевой переменной. Данный столбец подлежит удалению.